[WIP] Add support for access keys

thomas4019 opened the pull request:
The main thing missing is some api endpoint for creating access keys.

kane-mason commented on modules/access_keys/access_keys.js:
```diff
@@ -0,0 +1,30 @@
+      type: 'object',
+      additionalProperties: false,
+      properties: {
+        user_id: {
```
with the more recent changes you will need to add property `user_collection` to cater for users that are not necessarilly `users`

kane-mason commented on auth/index.js:
```diff
@@ -17,6 +17,32 @@ exports.isValidPassword = function (password, hashedPassword) {
+        if (accessKey.expires_at >= new Date().toISOString()) {
+          let user
+          try {
+            user = await req.db['users'].get(accessKey.user_id)
```
once `user_collection` added to schema this will become `req.db[accessKey.user_collection]`

kane-mason commented on auth/index.js:
```diff
@@ -17,6 +17,32 @@ exports.isValidPassword = function (password, hashedPassword) {
+  const key = req.query['access_key'] || req.headers['x-access-key']
+  if (key) {
+    if (typeof key == 'string') {
+      const accessKeys = await req.db['access_keys'].find({ key: { $eq: key } })
```
why dont we short circuit this to only return top 1 result

kane-mason commented:
> The main thing missing is some api endpoint for creating access keys.

well the standard listeners will exist right? Or do you mean creating an expressa internal listener to populate access key properties?

kane-mason commented:
> The main thing missing is some api endpoint for creating access keys.

Will these not be added using default collection endpoints? Then we can create in admin?

kane-mason commented:
@thomas4019 would love to pick this up again and get it in

thomas4019 commented:
@kane-mason Sorry about the delay here. I've merged this in after addressing your comments.

Yeah, you're right that the standard admin endpoints are going to be the way to go for adding, editing and deleting access keys.
